### PR TITLE
feat: reformatted log2 function

### LINKED ISSUE #23765: log2

### PR DESCRIPTION
**Pull request by Mr-Array22**
# PR Description

Update the log2 function

## Related Issue

Close #23765

## Checklist

- [ ] Did you add a function?
- [ ] Did you add the tests?
- [ ] Did you run your tests and are your tests passing?
- [ ] Did pre-commit not fail on any check?
- [✓] Did you follow the steps we provided?

### Socials:
@Mostafa_Gemy22

### CODE REVIEW

File: ivy/functional/ivy/elementwise.py [hunk 1]
```diff
@@ -4179,7 +4179,7 @@ def log1p(
 @handle_array_function
 @handle_device_shifting
 def log2(
-    x: Union[ivy.Array, ivy.NativeArray],
+    x: Union[ivy.Array, ivy.NativeArray, Number],
```
Comment (Dsantra92) [inline ivy/functional/ivy/elementwise.py:4182]: Any reason why Number was added as supported type?

**Review by Dsantra92 — CHANGES_REQUESTED** [review]
The formatting in ivy api LGTM except the type annotation. Also, the .idea files shouldn't be committed.

File: ivy/functional/ivy/elementwise.py (hunk 1 above)
Comment (Mr-Array22) [reply ivy/functional/ivy/elementwise.py:4182]: Sorry for this  mistake ,I will fix it

Comment (Mr-Array22) [reply ivy/functional/ivy/elementwise.py]: @Dsantra92  i fix it now ..can you check it again?

**Pull request closed without merging**